pkg/pkgrepository/app_deploy.go

```diff
@@ -30,13 +30,105 @@ func (a *App) deploy(tplOutput string) exec.CmdRunResult {
+		fmt.Println("sending to kapp deploy: \n", appendRebaseRule(tplOutput))
+		return kapp.Deploy(appendRebaseRule(tplOutput), a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
 
 	default:
 		return exec.NewCmdRunResultWithErr(fmt.Errorf("Unsupported way to deploy"))
 	}
 }
 
+func appendRebaseRule(tplOutput string) string {
+	return tplOutput + `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+- ytt:
+    overlayContractV1:
+      overlay.yml: |
+        #@ load("@ytt:data", "data")
+        #@ load("@ytt:json", "json")
+        #@ load("@ytt:overlay", "overlay")
+        #@ load("@ytt:struct", "struct")
+
+        #@ def get_rev(annotations):
+        #@  if hasattr(annotations, "packaging.carvel.dev/repo-rev"):
+        #@    return int(annotations["packaging.carvel.dev/repo-rev"])
+        #@  else:
+        #@    return -1
+        #@  end
+        #@ end
+        #@ if not hasattr(data.values.existing.metadata.annotations, "packaging.carvel.dev/package-repository-ref"):
+        #@   msg = "Error: cannot overwrite package " + data.values.existing.metadata.name + " because it was not created by a package repository."
+        #@   print(msg)
+        #@   fail(msg)
+        #@ end
+
+        #@ def annotations_are_identical(existing_anns, new_anns):
+        #@   carvel_anns = {"kapp.k14s.io/disable-original": True,
+        #@                   "kapp.k14s.io/disable-wait": True,
+        #@                   "packaging.carvel.dev/package-repository-ref": True,
+        #@                   "kapp.k14s.io/identity": True,
+        #@                   "kapp.k14s.io/create-strategy": True}
+        #@   return {x: v for x,v in existing_anns.items() if x not in carvel_anns} == {x: v for x,v in new_anns.items() if x not in carvel_anns}
+        #@ end
+
+        #@ def labels_are_identical(existing_labels, new_labels):
+        #@   carvel_labels = {"kapp.k14s.io/app": True, "kapp.k14s.io/association": True}
+        #@   return {x: v for x,v in existing_labels.items() if x not in carvel_labels} == {x: v for x,v in new_labels.items() if x not in carvel_labels}
+        #@ end
+
+        #@ def is_identical(existing, new):
+
+        #@   print( "\nrefnames? " + str(existing.spec.refName == new.spec.refName) + \
+        #@           "\nversion? " + str(existing.spec.version == new.spec.version) + \
+        #@           "\ntemplate? " + str(struct.decode(existing.spec.template) == struct.decode(new.spec.template)) + \
+        #@           "\nlabels? " + str(labels_are_identical(struct.decode(existing.metadata.labels), struct.decode(new.metadata.labels))) +\
+        #@           "\nannotations? " + str(annotations_are_identical(struct.decode(existing.metadata.annotations), struct.decode(new.metadata.annotations))))
+
+        #@ print ("\n existing template: \n" + str(struct.decode(existing.spec.template)) + "\n new template: " + str(struct.decode(new.spec.template)))
```

Comment:
these are debug prints,  but imo might be helpful to thread a debug flag through to here so that when someone gets stuck they're able to ask "why" in a meaningful way --- any thoughts on how best to enable introspection / debugging @cppforlife  ?